lib/active_model/serializer/caching.rb

```diff
@@ -42,7 +43,12 @@ module ClassMethods
         def inherited(base)
           super
           caller_line = caller[1]
-          base._cache_digest = digest_caller_file(caller_line)
+          base._cache_digest_file_path = caller_line
+        end
+
+        def _cache_digest
+          return @_cache_digest if defined?(@_cache_digest)
+          @_cache_digest = digest_caller_file(_cache_digest_file_path)
```

Comment:
You're right. It should never be nilnor empty
On Mon, Apr 18, 2016 at 6:04 PM Lucas Hosseini <notifications@github.com>
wrote:

> In lib/active_model/serializer/caching.rb
> <https://github.com/rails-api/active_model_serializers/pull/1687#discussion_r60149660>
> :
>
> > @@ -42,7 +43,12 @@ module ClassMethods
> >          def inherited(base)
> >            super
> >            caller_line = caller[1]
> > -          base._cache_digest = digest_caller_file(caller_line)
> > +          base._cache_digest_file_path = caller_line
> > +        end
> > +
> > +        def _cache_digest
> > +          return @_cache_digest if defined?(@_cache_digest)
> > +          @_cache_digest = digest_caller_file(_cache_digest_file_path)
>
> Could be rewritten as @_cache_digest ||=
> digest_caller_file(_cache_digest_file_path) instead of the two lines.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/rails-api/active_model_serializers/pull/1687/files/1d24c9708ae802ec449664f82746d93d8fa345c0#r60149660>
>